frontend/src/routes/Applications/Overview.tsx

```diff
@@ -440,10 +442,18 @@ export default function ApplicationsOverview() {
     return fetchAggregate(SupportedAggregate.applications, backendUrl, requestedExport)
   }
 
-  const tableItems: IResource[] = useMemo(
-    () => [...allApplications.map((app) => generateTransformData(app))],
-    [allApplications, generateTransformData]
-  )
+  const tableItems: IResource[] = useMemo(() => {
+    const items = allApplications
+    /* istanbul ignore next */
+    deletedApps.forEach((dapp) => {
+      const inx = items.findIndex((app) => dapp.metadata?.uid === app.metadata?.uid)
+      if (inx !== -1) {
+        items.splice(inx, 1)
+        resultCounts.itemCount -= 1
```

Comment:
Filter counts will be wrong until the app stops being returned by the backend. How long do you suppose that can take?

I was thinking we should clear out any deleted apps if we don't find them in the list, but then I realized these are only stored in state. So navigating away and coming back to the page, the app might reappear right?